### PR TITLE
Stats: fix stats refresh when day changes in background

### LINKED ISSUE #19688: [iOS] Stats: Days tab doesn't display today's data after backgrounding the app and returning the next day
### Expected behavior

Stats days view should reload after coming back to the app the next day and correctly display today's data in labels and graph.

Related issue: https://github.com/wordpress-mobile/WordPress-iOS/issues/18260

### Actual behavior

Stats days view shows 0s and an outdated graph after coming back to the app the next day. Pull to refresh doesn't help.

**Workaround:** Switch between "Weeks and Days" tab

### Steps to reproduce the behavior

1. Set iPhone clock to yesterday
2. Fresh install WordPress and login to account that has stats
3. Open Stats
4. Open Days tab
5. Background the app
6. Set iPhone clock back to today
7. Come back to Stats
8. Notice that yesterday's date is selected
9. Select today by clicking ">" icon
10. Notice 0s are shown and the graph still displays yesterday's data
11. Try pull to refresh
12. Notice that the view still didn't refresh
13. Switch between the "Weeks" and "Days" tab
14. Notice that now the "Days" tab displays today's data correctly

https://user-images.githubusercontent.com/4062343/204784611-223ecfdd-346b-4192-9306-6f6e05f53aae.mov

### PR DESCRIPTION
**Pull request by sla8c**
This PR fixes an issue where stats were not updating correctly when the date incremented on iOS in the background

Fixes #19688

To test:
1. Set iPhone clock to yesterday on iOS via Settings
2. Fresh install JetPack app and login to account that has stats
3. Open Stats
4. Tap the Days tab
5. Background the JetPack app
6. Set iPhone clock back to today on iOS via Settings
7. Relaunch the JetPack app which should still be on the Stats -> Days tab screen
8. Notice that yesterday's date is selected
9. Select today by clicking ">" icon
10. Notice that now the "Days" tab displays today's data correctly

## Regression Notes
1. Potential unintended areas of impact
Stats

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Stats/Period Stats/SiteStatsPeriodViewModel.swift [hunk 1]
```diff
@@ -287,7 +287,12 @@ private extension SiteStatsPeriodViewModel {
             mostRecentChartData.periodEndDate == periodSummary.periodEndDate {
             self.mostRecentChartData = periodSummary
         } else if let periodSummary = periodSummary, let chartData = mostRecentChartData, periodSummary.periodEndDate > chartData.periodEndDate {
-            mostRecentChartData = chartData
+            if let lastSummaryDataEntry = summaryData.last, periodSummary.periodEndDate == lastSummaryDataEntry.periodStartDate {
```
Comment (staskus) [inline WordPress/Classes/ViewRelated/Stats/Period Stats/SiteStatsPeriodViewModel.swift]: I wonder if we could somehow improve the readability of all of these ifs. Without context, it takes a lot of time to understand these checks.

The easiest option I see is to make short comments above conditional statements. The harder option may be extracting the checks into methods.

WDYT?

File: WordPress/Classes/ViewRelated/Stats/Period Stats/SiteStatsPeriodViewModel.swift [hunk 2]
```diff
@@ -287,7 +287,12 @@ private extension SiteStatsPeriodViewModel {
             mostRecentChartData.periodEndDate == periodSummary.periodEndDate {
             self.mostRecentChartData = periodSummary
         } else if let periodSummary = periodSummary, let chartData = mostRecentChartData, periodSummary.periodEndDate > chartData.periodEndDate {
```
Comment (staskus) [inline WordPress/Classes/ViewRelated/Stats/Period Stats/SiteStatsPeriodViewModel.swift]: I wonder if these checks can be simplified or if some of them are not needed:

The last one set `chartData` to `mostRecentChartData` and then set `mostRecentChartData` to this `chartData`

```
else if <...> let chartData = mostRecentChartData <...> {
    mostRecentChartData = chartData
}
```

**Review by staskus — APPROVED** [review]
Thank you for the fix, the solution works!

I added some additional comments regarding code clarity. However, I don't have a concrete proposal and some of it concerns previously written code as well.

File: WordPress/Classes/ViewRelated/Stats/Period Stats/SiteStatsPeriodViewModel.swift (hunk 1 above)
Comment (sla8c) [reply WordPress/Classes/ViewRelated/Stats/Period Stats/SiteStatsPeriodViewModel.swift]: @staskus sounds good I made comments on the conditional statement that I touched which in my mind was the least transparent and most unclear on the intended behaviour

File: WordPress/Classes/ViewRelated/Stats/Period Stats/SiteStatsPeriodViewModel.swift (hunk 2 above)
Comment (sla8c) [reply WordPress/Classes/ViewRelated/Stats/Period Stats/SiteStatsPeriodViewModel.swift]: @staskus I had thought this myself but I didn't want to introduce a bug + not sure what the intention was so I took your first suggestion and only did comments for now

**Review by staskus — APPROVED** [review]

**Pull request merged**